### PR TITLE
DB changes  for user details

### PR DESCRIPTION
**Pull request by RiteshKolhe**
Captured the qualification and vehicle.

### CODE REVIEW

File: ePanchayat.DB/Tables/DDL/House_tbl.sql [hunk 1]
```diff
@@ -1,24 +1,25 @@
...
+CREATE TABLE dbo.House_tbl
+(
+	HouseId INT IDENTITY(1,1) NOT NULL,
+	HouseNumber VARCHAR(50) NOT NULL,
+	OwnerId INT NOT NULL,
+	Landmark VARCHAR(50) NOT NULL,
+	LastModifiedOn DATETIME NOT NULL,
+	LastModifiedBy VARCHAR(50) NOT NULL,
+	IsActive BIT NOT NULL
+)
```
Comment (mohandivraniya) [inline ePanchayat.DB/Tables/DDL/House_tbl.sql:24]: Let we please add all related constraint for Primary key, foreign key, default etc.

File: ePanchayat.DB/Tables/DDL/Qualification_tbl.sql [hunk 2]
```diff
@@ -1,24 +1,26 @@
...
+(
+	QualificationId INT IDENTITY(1,1) NOT NULL,
+	UserId INT NOT NULL,
+	Qualification VARCHAR(50) NOT NULL,
+	Major VARCHAR(50) NOT NULL,
+	PassingYear INT NULL,
+	LastModifiedOn DATETIME NOT NULL,
+	LastModifiedBy VARCHAR(50) NOT NULL,
+	IsActive BIT NOT NULL
+)
```
Comment (mohandivraniya) [inline ePanchayat.DB/Tables/DDL/Qualification_tbl.sql:25]: Same here - please add all required constraints

File: ePanchayat.DB/StoredProcedures/HouseGet_sp.sql [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+	SELECT
+		VehicleId
+		,Category
+		,RegistrationNumber
+		,OwnerId
+		,LastModifiedOn
+		,LastModifiedBy
+		,IsActive
+	FROM
+		dbo.House_tbl
```
Comment (mohandivraniya) [inline ePanchayat.DB/StoredProcedures/HouseGet_sp.sql:25]: Let we perform join with user_tbl to get user details also for Owner and ModifiedBy user.

File: ePanchayat.DB/StoredProcedures/HouseSave_sp.sql [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+IF OBJECT_ID('dbo.HouseSave_sp') IS NOT NULL
+BEGIN
+	DROP PROCEDURE dbo.HouseSave_sp
+END
+GO
+
+CREATE PROCEDURE dbo.HouseSave_sp
+(
+	@HouseId INT
```
Comment (mohandivraniya) [inline ePanchayat.DB/StoredProcedures/HouseSave_sp.sql:12]: Fields can be nullable while doing insert, Also fields like Landmark will null

File: ePanchayat.DB/StoredProcedures/QualificationGet_sp.sql [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+		QualificationId
+		,UserId
+		,Qualification
+		,Major
+		,PassingYear
+		,LastModifiedOn
+		,LastModifiedBy
+		,IsActive
+	FROM
+		dbo.Qualification_tbl
```
Comment (mohandivraniya) [inline ePanchayat.DB/StoredProcedures/QualificationGet_sp.sql:27]: Same here - Perform join while fetching data

File: ePanchayat.DB/StoredProcedures/VehicleGet_sp.sql [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+	SELECT
+		VehicleId
+		,Category
+		,RegistrationNumber
+		,OwnerId
+		,LastModifiedOn
+		,LastModifiedBy
+		,IsActive
+	FROM
+		dbo.Vehicle_tbl
```
Comment (mohandivraniya) [inline ePanchayat.DB/StoredProcedures/VehicleGet_sp.sql:26]: Perform join to fetch user details

File: ePanchayat.DB/StoredProcedures/VehicleSave_sp.sql [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+
+IF OBJECT_ID('dbo.VehicleSave_sp') IS NOT NULL
+BEGIN
+	DROP PROCEDURE dbo.VehicleSave_sp
+END
+GO
+
+CREATE PROCEDURE dbo.VehicleSave_sp
+(
+	@VehicleId INT
```
Comment (mohandivraniya) [inline ePanchayat.DB/StoredProcedures/VehicleSave_sp.sql:12]: Fields can be nullable

**Review by mohandivraniya — CHANGES_REQUESTED** [review]
Please address issues:
1. Add constraints to manage data efficiently
2. Include Joins while fetching details
3. Address nullable fields

**Comment by RiteshKolhe** [comment]
> Captured the qualification and vehicle.

Addressed the issue

**Comment by RiteshKolhe** [comment]
> Captured the qualification and vehicle.

It's done

**Review by mohandivraniya — APPROVED** [review]

**Pull request merged**